website/blog/2020-11-06-sbt-BSP-support.md

```diff
@@ -0,0 +1,147 @@
+    Bloop.
+- Bloop writes all the necessary information about your build it needs to disk
+    whereas sbt loads it into memory. If you're a user that works for long
+    periods of time on a single project, then this may not matter to you at all.
+    However, if you jump in and out of projects, without having sbt shell
+    running, then you will pay the cost of loading up your build every time.
+- Bloop offers sbt file support for completions and hover, whereas this is not
+    yet available in the sbt BSP implementation.
+- sbt gives you access to the full task graph, so for example if you're using
+    `BuildInfo`, and you compile with Bloop, your `BuildInfo` won't get
+    generated. However, sources will get generated when using sbt server. This
+    can help avoid any potential inconsistencies. 
+
+While there are some more differences, these are probably the main ones you'll
+notice as a user. These are both great tools, and you'll have to explore more to
+see what is the best fit for you. For now, Metals defaults to using Bloop mainly
+because we believe it provides a richer feature set and because pretty much all
+of our testing includes Bloop as a build server.
+
+## How do I try it out?
+
+Now to the good stuff. There are two main ways that you can start using sbt
+server in Metals.
+
+### No `.bsp/sbt.json` exists
+
+If no `.bsp/sbt.json` exists yet, you can generate it the new
+`metals.generate-bsp-config` command. At any time you can execute this command
+and Metals will ensure that you're on the minimum required sbt version (1.4.1),
+generate the `.bsp/sbt.json` file for you, include the necessary plugin, and
+then auto-connect to the sbt build server. For example in VS Code this looks
+like this:
+
+![generate-bsp-config](https://i.imgur.com/kBNbtzI.gif)
+
+### `.bsp/sbt.json` already exists
+
+If you've ran sbt on your project already, your `.bsp/sbt.json` file will
+already exist. In this case, you can simply use the `metals.bsp-switch` command
+and choose sbt. For example, using `coc-metals`, it looks like this:
+
+![bsp-switch](https://i.imgur.com/6tY2ofL.gif)
+
+### Switching back to Bloop
+
+After you do a `metals.bsp-switch` or `metals-generate-bsp-config`, your choice
+will be remembered. So the next time that you open your workspace, you will
+automatically connect to the last build server you had chosen. If you'd like to
+go back to your previous build server, you can simply use the
+`metals.bsp-switch` command again to choose your previous server. For editors
+that have an html doctor, you can also reset your build server choice in the
+Doctor.
+
+![Doctor](https://i.imgur.com/YEGfEGB.png)
+
+## Conclusion
+
+We hope you enjoy this easier way to use sbt server with Metals. If you're
+curious about the implementation details, much of the work was done in [this
+pr](https://github.com/scalameta/metals/pull/2154).  As always please don't
+hesitate to ask questions on our various channels, submit issues, or create new
```

Comment:
```suggestion
hesitate to ask questions on our various channels, submit issues or create new
```

I did comment on this in a couple of places, but maybe it's my bad knowledge of the English language. Do we need a coma before `and` or `or` ?

Reply:
Welcome to the oxford comma debate. https://www.grammarly.com/blog/oxford-comma-debate/

In short, both are correct 😆

Reply:
Och no... :(